app/shared/src/commonMain/kotlin/ui/subject/episode/details/DanmakuListSection.kt

```diff
@@ -0,0 +1,379 @@
+import androidx.compose.material3.CircularProgressIndicator
+import androidx.compose.material3.DropdownMenu
+import androidx.compose.material3.DropdownMenuItem
+import androidx.compose.material3.FilterChip
+import androidx.compose.material3.FilterChipDefaults
+import androidx.compose.material3.Icon
+import androidx.compose.material3.ListItem
+import androidx.compose.material3.ListItemDefaults
+import androidx.compose.material3.MaterialTheme
+import androidx.compose.material3.Surface
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.runtime.collectAsState
+import androidx.compose.runtime.derivedStateOf
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.text.style.TextOverflow
+import androidx.compose.ui.unit.dp
+import kotlinx.coroutines.flow.Flow
+import me.him188.ani.app.domain.episode.DanmakuFetchResultWithConfig
+import me.him188.ani.danmaku.api.DanmakuServiceId
+import me.him188.ani.danmaku.api.provider.DanmakuMatchMethod
+import me.him188.ani.danmaku.ui.DanmakuPresentation
+
+/**
+ * 弹幕列表项数据类，用于在UI中显示单条弹幕信息。
+ * 
+ * @property id 弹幕唯一标识符
+ * @property content 弹幕文本内容
+ * @property timeMillis 弹幕出现时间（毫秒）
+ * @property serviceId 弹幕来源服务ID
+ * @property isSelf 是否为当前用户发送的弹幕
+ */
+data class DanmakuListItem(
+    val id: String,
+    val content: String,
+    val timeMillis: Long,
+    val serviceId: DanmakuServiceId,
+    val isSelf: Boolean,
+)
+
+/**
+ * 弹幕列表区域组件，提供弹幕源选择和弹幕列表显示功能。
+ *
+ */
+@OptIn(ExperimentalLayoutApi::class)
+@Composable
+fun DanmakuListSection(
+    danmakuFlow: Flow<List<DanmakuPresentation>>,
+    fetchResults: List<DanmakuFetchResultWithConfig>,
+    expanded: Boolean,
+    onToggleExpanded: () -> Unit,
+    onSetEnabled: (DanmakuServiceId, Boolean) -> Unit,
+    onManualMatch: (DanmakuServiceId) -> Unit,
+    modifier: Modifier = Modifier,
+) {
+    val danmakuList by danmakuFlow.collectAsState(initial = emptyList())
+    
+    // 获取所有可用源
+    val availableSources = remember(fetchResults) {
+        fetchResults.map { it.serviceId }.toSet()
+    }
+    
+    // 默认全部启用
+    var selectedSources by remember(availableSources) {
+        mutableStateOf(availableSources)
+    }
+    
+    // 初始化时启用所有源
+    LaunchedEffect(availableSources) {
+        if (availableSources.isNotEmpty()) {
+            selectedSources = availableSources
+            availableSources.forEach { serviceId ->
+                onSetEnabled(serviceId, true)
+            }
+        }
+    }
+    
+    // 显示弹幕列表，注意这里的弹幕已经是经过 [EpisodeDanmakuLoader] 过滤的
+    val filteredDanmaku by remember(danmakuList) {
+        derivedStateOf {
+            danmakuList.map { presentation ->
```

Comment:
`remember(danmakuList)` 和 `derivedStateOf` 是'冲突'的,


```suggestion
    val filteredDanmaku by remember {
        derivedStateOf {
            danmakuList.map { presentation ->
```